benchmarks/perf-tool/okpt/test/steps/steps.py

```diff
@@ -520,16 +744,20 @@ def recall_at_r(results, neighbor_dataset, r, k, query_count):
         Recall at R
     """
     correct = 0.0
+    total_num_of_results = 0
     for query in range(query_count):
         true_neighbors = neighbor_dataset.read(1)
         if true_neighbors is None:
             break
         true_neighbors_set = set(true_neighbors[0][:k])
-        for j in range(r):
+        true_neighbors_set.discard(-1)
```

Comment:
Why do we need to do this?

Reply:
this way we use all neighbors after filter has been applied. 
I faced some issues with hdf5 format of variable length, so I'm using same length list for each query result. As we don't know number of neighbors beforehand we pre-populate list with -1 and then treat -1 as null value.
For example, dataset has 5 true neighbors [5, 2, 1, 4, 0], after applying filter we have only 3 [2, 1, 0]. When we read this list to compute recall we'll get  [2, 1, 0, -1, -1] and after `true_neighbors_set.discard(-1)` it will be translated to [2, 1, 0].